template/src/services/api.ts

```diff
@@ -0,0 +1,22 @@
+import axios from 'axios';
+
+export const api = axios.create({
+  baseURL: process.env.REACT_APP_API_URL,
+});
+
+api.defaults.timeout = 10000000;
```

Comment:
Why this timeout?

Reply:
What kind of improvement/benefit does this bring to the application? I don't see any.

Reply:
It's an axios config (optional of course). I think the benefit is that if the future API request is getting too much time (pending status), and for a reason doesn't throw a timeout, the client side can close the connection throwing the Client Side timeout. And as it is just a configuration, I don't know if we lose something by not configurating it, but I think eventually it can helps us .

Reply:
The first paragraph of this [article](https://medium.com/@masnun/handling-timeout-in-axios-479269d83c68) tell us a little bit about it.

Reply:
Got it! I just want to point out some notes:

- I don't think the user will wait 3 minutes (neither 2 nor 1 minute) to get any feedback on the request (status aborted). Before that, he'll probably have reloaded the page a few times. So, that config ends up becoming irrelevant.
- Also, in this waiting time, the user can have navigated to another page in the app. If the promise comes to be resolved, it will be executed in an unmounted component and will throw an error, which can lead to memory leaks. So, I think the best way to solve this would be to cancel/abort the request on the "component will unmount" hook (this could be a future implementation in the application).
- If we set a shorter time, this can lead to false-positive feedback, i.e., the timeout time may be less than the (real) request's response, although unlikely, but that can happen and bring some inconsistencies with the API data.

Anyway, it's ok to leave this configuration as it is for now, but I would remove this configuration because I don't see any real advantage in using it in our context.

Reply:
Yess, I agree with all those points.